PronamicsToyRobot/Classes/Controller.cs

```
@@ -0,0 +1,99 @@
+﻿using System;
+
+namespace PronamicsToyRobot.Classes
+{
+    /// <summary>
+    /// The controller class is where the user's input is filtered to, but it 
+    /// is also the monitor of the board and robot. 
+    /// 
+    /// The reason for this is due for extensibility: should we want the robot 
+    /// to pick up or put down objects, or have multiple robots, the controller
+    /// would be a good starting point for tracking multiple objects. 
+    /// </summary>
+    public class Controller
+    {
+        private Robot myRobot;
+        private Board myBoard;
+
+
+        public Action<string> ReportAction { get; set; } 
+
+        public Controller() { 
+            myRobot = new Robot();
+            myBoard = Board.CreateDefaultBoard();
+        }
+
+        public void CommandRobot(string command)
+        {
+            if (string.IsNullOrWhiteSpace(command))
+            {
+                Report("Ignored");
+                return;
+            }
+            string[] commandSplit = command.Split(' ');
+            string commandKeyword = commandSplit[0];
+
+            switch (commandKeyword)
+            {
+                case "PLACE":
+                    if (commandSplit.Length > 1)
+                        PlaceRobot(commandSplit[1]);
+                    break;
+                case "MOVE":
+                    MoveRobot();
+                    break;
+                case "RIGHT":
+                    myRobot.Turn(true);
+                    break;
+                case "LEFT":
+                    myRobot.Turn(false);
+                    break;
+                case "REPORT":
+                    Report(myRobot.Report());
+                    break;
+                default:
```

If it doesn't somewhere else, this should be notifying the user rather than silently failing. Informing the user of what they've done wrong so that they can fix the problem is key to good user interactions.